library/std/src/collections/hash/map.rs

```diff
@@ -2501,6 +2502,58 @@ where
 }
 
 impl<'a, K, V> Entry<'a, K, V> {
+    /// Inserts `value` if the entry is vacant, otherwise merges the existing
+    /// value with `value`, using the given function `f`.
+    ///
+    /// If a panic occurs in the function, the entry will be removed.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(hashmap_entry_merge)]
+    ///
+    /// use std::collections::HashMap;
+    /// use std::ops::Add;
+    ///
+    /// let mut word_counts: HashMap<&str, u32> = HashMap::new();
+    /// word_counts.entry("very").merge(1, Add::add);
+    /// word_counts.entry("very").merge(1, Add::add);
+    /// word_counts.entry("wonderful").merge(1, Add::add);
+    ///
+    /// assert_eq!(word_counts["very"], 2);
+    /// assert_eq!(word_counts["wonderful"], 1);
+    /// ```
+    #[unstable(feature = "hashmap_entry_merge", issue = "none")]
+    pub fn merge(self, value: V, f: impl FnOnce(V, V) -> V) -> &'a mut V {
```

Comment:
- I think generally `impl Trait` syntax is avoided in argument position of public methods because it prevents users from specifying the concrete generic, although I see it was used in `Peekable::next_if` (and only there it seems)
- What if I don't want to create a new `V` when the entry is an `OccupiedEntry`? I guess it's no longer a "merge", but fundamentally it requires the same operation.